tests/python_package_test/test_dask.py

```diff
@@ -198,7 +198,7 @@ def _create_data(objective, n_samples=1_000, output='array', chunk_size=500, **k
 
 def _r2_score(dy_true, dy_pred):
     numerator = ((dy_true - dy_pred) ** 2).sum(axis=0, dtype=np.float64)
-    denominator = ((dy_true - dy_pred.mean(axis=0)) ** 2).sum(axis=0, dtype=np.float64)
+    denominator = ((dy_true - dy_true.mean(axis=0)) ** 2).sum(axis=0, dtype=np.float64)
```

Comment:
I expect the other changes in this PR to generate a bit of discussion, given that you said you saw other tests failing as a result of them. Please make this `_r2_score()` fix a separate PR so it doesn't get stuck waiting on that discussion to be resolved.

Reply:
fixed in #4151